third_party/cronet/cronet.idl_c.h

```diff
@@ -0,0 +1,1288 @@
+  Cronet_Error_ERROR_CODE_ERROR_QUIC_PROTOCOL_FAILED = 10,
+  Cronet_Error_ERROR_CODE_ERROR_OTHER = 11,
+} Cronet_Error_ERROR_CODE;
+
+typedef enum Cronet_EngineParams_HTTP_CACHE_MODE {
+  Cronet_EngineParams_HTTP_CACHE_MODE_DISABLED = 0,
+  Cronet_EngineParams_HTTP_CACHE_MODE_IN_MEMORY = 1,
+  Cronet_EngineParams_HTTP_CACHE_MODE_DISK_NO_HTTP = 2,
+  Cronet_EngineParams_HTTP_CACHE_MODE_DISK = 3,
+} Cronet_EngineParams_HTTP_CACHE_MODE;
+
+typedef enum Cronet_UrlRequestParams_REQUEST_PRIORITY {
+  Cronet_UrlRequestParams_REQUEST_PRIORITY_REQUEST_PRIORITY_IDLE = 0,
+  Cronet_UrlRequestParams_REQUEST_PRIORITY_REQUEST_PRIORITY_LOWEST = 1,
+  Cronet_UrlRequestParams_REQUEST_PRIORITY_REQUEST_PRIORITY_LOW = 2,
+  Cronet_UrlRequestParams_REQUEST_PRIORITY_REQUEST_PRIORITY_MEDIUM = 3,
+  Cronet_UrlRequestParams_REQUEST_PRIORITY_REQUEST_PRIORITY_HIGHEST = 4,
+} Cronet_UrlRequestParams_REQUEST_PRIORITY;
+
+typedef enum Cronet_UrlRequestParams_IDEMPOTENCY {
+  Cronet_UrlRequestParams_IDEMPOTENCY_DEFAULT_IDEMPOTENCY = 0,
+  Cronet_UrlRequestParams_IDEMPOTENCY_IDEMPOTENT = 1,
+  Cronet_UrlRequestParams_IDEMPOTENCY_NOT_IDEMPOTENT = 2,
+} Cronet_UrlRequestParams_IDEMPOTENCY;
+
+typedef enum Cronet_RequestFinishedInfo_FINISHED_REASON {
+  Cronet_RequestFinishedInfo_FINISHED_REASON_SUCCEEDED = 0,
+  Cronet_RequestFinishedInfo_FINISHED_REASON_FAILED = 1,
+  Cronet_RequestFinishedInfo_FINISHED_REASON_CANCELED = 2,
+} Cronet_RequestFinishedInfo_FINISHED_REASON;
+
+typedef enum Cronet_UrlRequestStatusListener_Status {
+  Cronet_UrlRequestStatusListener_Status_INVALID = -1,
+  Cronet_UrlRequestStatusListener_Status_IDLE = 0,
+  Cronet_UrlRequestStatusListener_Status_WAITING_FOR_STALLED_SOCKET_POOL = 1,
+  Cronet_UrlRequestStatusListener_Status_WAITING_FOR_AVAILABLE_SOCKET = 2,
+  Cronet_UrlRequestStatusListener_Status_WAITING_FOR_DELEGATE = 3,
+  Cronet_UrlRequestStatusListener_Status_WAITING_FOR_CACHE = 4,
+  Cronet_UrlRequestStatusListener_Status_DOWNLOADING_PAC_FILE = 5,
+  Cronet_UrlRequestStatusListener_Status_RESOLVING_PROXY_FOR_URL = 6,
+  Cronet_UrlRequestStatusListener_Status_RESOLVING_HOST_IN_PAC_FILE = 7,
+  Cronet_UrlRequestStatusListener_Status_ESTABLISHING_PROXY_TUNNEL = 8,
+  Cronet_UrlRequestStatusListener_Status_RESOLVING_HOST = 9,
+  Cronet_UrlRequestStatusListener_Status_CONNECTING = 10,
+  Cronet_UrlRequestStatusListener_Status_SSL_HANDSHAKE = 11,
+  Cronet_UrlRequestStatusListener_Status_SENDING_REQUEST = 12,
+  Cronet_UrlRequestStatusListener_Status_WAITING_FOR_RESPONSE = 13,
+  Cronet_UrlRequestStatusListener_Status_READING_RESPONSE = 14,
+} Cronet_UrlRequestStatusListener_Status;
+
+// // Declare constants
+
+// ///////////////////////
+// // Concrete interface Cronet_Buffer.
+
+// // Create an instance of Cronet_Buffer.
+CRONET_EXPORT Cronet_BufferPtr Cronet_Buffer_Create(void);
+// // Destroy an instance of Cronet_Buffer.
+// CRONET_EXPORT void Cronet_Buffer_Destroy(Cronet_BufferPtr self);
```

Comment:
Don't comment out code in this copied source file. Instead, use ffigen to filter the method names if you don't want to generate all of them.